test/blackbox-tests/test-cases/pkg/lockfile-generation.t/run.t

```diff
@@ -119,6 +127,14 @@ Run the solver again preferring oldest versions of dependencies:
    (complete false)
    (used))
   
+  (solver_env
+   (flags
+    ("with-doc" "with-test"))
```

Comment:
do these need quoting?

Reply:
Yes because it's an OSL and so if the first term is a string literal it must be quoted. Subsequent values are quoted for consistency.

Reply:
But why is it OSL? I expected that we would expand the OSL to an actual ordered set when writing the flags here.

Reply:
But then `encode` and `decode` wouldn't round-trip, and the syntax for the solver env in the lockdir won't match the syntax used for the solver env in the workspace config.